### PR TITLE
fix(analytics): Room name persistant prop.

### PR DESCRIPTION
**Pull request by hristoterezov**

### CODE REVIEW

**Comment by virtuacoplenny** [comment]
Could you briefly tell me the timing issue. setRoom was called after libInit?

**Comment by saghul** [comment]
But SET_ROOM is called for for every room you join on mobile, so this would re-initialize analytics over and over again. Is this the desired behavior?

**Comment by saghul** [comment]
So when do you need it exactly? Wouldn’t waiting for all middlewares / reducers to run after LIB_DID_INIT suffice? If so, you can do it after next(action)

**Comment by hristoterezov** [comment]
@saghul I'm not sure I understand your comment. The problem that the PR is trying to solve is the following:
In the initAnalytics we are setting the room name as permanent property. Obviously if we execute  initAnalytics before SET_ROOM action we wouldn't be able to get the room name from the store.

But since on mobile I should expect multiple SET_ROOM actions to be dispatched I have to rethink how to handle this. Also I suspect that something was wrong even with the initial implementation but I'll check this.

**Comment by hristoterezov** [comment]
@saghul I checked again and it should be fine!

On mobile any time when we are entering the room we are fetching the config ( which is necessary because the deployment can be different ). Which basically means that potentially we may need new handlers and we need to fetch them and execute initAnalytics again. 

But even before this PR on mobile we were executing setConfig every time when the user is entering the room which will trigger initLib action which will result in initAnalytics. So even before the PR, every time when the user joins a room we were executing initAnalytics.

I check the code again and IMHO there is no problem to init the analytics every time when the user joins the room and we need this for the use case where the user is using another deployment.

**Comment by virtuacoplenny** [comment]
Side question, no changes being requested: is there a reason why it's preferred to do something like this in middleware versus chained in the action or wherever calls the action?

**Comment by saghul** [comment]
> But even before this PR on mobile we were executing setConfig every time when the user is entering the room which will trigger initLib action which will result in initAnalytics. So even before the PR, every time when the user joins a room we were executing initAnalytics.

Hum, I would have sworn we don't reinitialize lib-jitsi-meet if the domain doesn't change... but I was wrong!

File: react/features/analytics/middleware.js [hunk 1]
```diff
@@ -12,9 +12,12 @@ import { initAnalytics } from './functions';
  */
 MiddlewareRegistry.register(store => next => action => {
     switch (action.type) {
-    case LIB_DID_INIT: {
+    case SET_ROOM: {
```
Comment (saghul) [inline react/features/analytics/middleware.js]: SET_ROOM will also happen when we navigate to the welcome page, and the room will be `undefined`. Do we also want to do this in that case?

**Comment by saghul** [comment]
> is there a reason why it's preferred to do something like this in middleware versus chained in the action or wherever calls the action?

The place where the `setRoom` action is dispatched is not aware of anything related to analytics. Analytics, however, is interested in doing *something* as a result of a SET_ROOM action, because it's only interested in knowing if there is a room or not.

Moreover, this would make more sense as a state listener. It's a new API we (mobile) introduced recently for attaching to redux state changes. In this case, you only want to reinitialize analytics when there is a change in the room, and when it's not undefined. Makes sense? See it in motion here: https://github.com/jitsi/jitsi-meet/pull/3050

File: react/features/analytics/middleware.js (hunk 1 above)
Comment (hristoterezov) [reply react/features/analytics/middleware.js]: We want to have send analytics for the welcome page and I don't think we can set a room. I think undefined is fine.

Comment (saghul) [reply react/features/analytics/middleware.js]: 👍

**Pull request merged**